lib/assessment.js

```diff
@@ -108,72 +108,27 @@ module.exports = {
     let updated;
     async.series(
       [
-        (callback) => {
-          sqldb.runInTransaction(
-            (client, callback) => {
-              debug('inside transaction');
-              async.series(
-                [
-                  (callback) => {
-                    sqldb.callWithClient(
-                      client,
-                      'assessment_instances_lock',
-                      [assessment_instance_id],
-                      (err) => {
-                        if (ERR(err, callback)) return;
-                        debug('locked');
-                        callback(null);
-                      }
-                    );
-                  },
-                  (callback) => {
-                    const params = [assessment_instance_id, authn_user_id];
-                    sqldb.callWithClientOneRow(
-                      client,
-                      'assessment_instances_update',
-                      params,
-                      (err, result) => {
-                        if (ERR(err, callback)) return;
-                        updated = result.rows[0].updated;
-                        debug('updated:', updated);
-                        callback(null);
-                      }
-                    );
-                  },
-                  (callback) => {
-                    if (!updated) return callback(null); // skip if not updated
+        async () => {
+          await sqldb.runInTransactionAsync(async (client) => {
+            await sqldb.callWithClientAsync(client, 'assessment_instances_lock', [
```

Comment:
In PT I've been relying on your code that uses AsyncLocalStorage to ensure that `callAsync()` inside `runInTransactionAsync` will automatically use the correct client, without needing to explicitly get the client and pass it down. Should we be promoting this as the preferred pattern?

Reply:
Yes! Good call; I'll make that change.

Reply:
Done in 6bfd6ae.